### PR TITLE
Feature/Update migration guide

### PR DESCRIPTION
**Pull request by Magnus-Kuhn**

### CODE REVIEW

File: _docs_integrate/version-5.md [hunk 1]
```diff
@@ -99,6 +99,69 @@ In addition, in contrast to the RequestItems, a RequestItemGroup did not have to
 The `mustBeAccepted` property of the RequestItems will be kept.
 {: .notice--info}
 
+## New mandatory wrappers for non-standard content of RelationshipTemplates, Relationships and Messages
```
Comment (jkoenig134) [inline _docs_integrate/version-5.md:102]: RelationshipCreationContent missing

File: _docs_integrate/version-5.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ required_by:
 
 The address format changed from `<3-character realm><32 or 33-character base58-string>` to `did:e:<backbone-hostname>:dids:<22-character lowercase hex string>`.
 
-## Removal of RelationshipChanges
+## Removal of RelationshipChanges {#relationshipchanges}
```
Comment (jkoenig134) [inline _docs_integrate/version-5.md:28]: why?

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

**Review by Magnus-Kuhn — COMMENTED** [review]
Some work still needs to be done after the remove-relationship-changes PR is merged

File: _docs_integrate/version-5.md (hunk 1 above)
Comment (Magnus-Kuhn) [reply _docs_integrate/version-5.md:102]: I wanted to have the sections as isolated as possible, so no creation content in the heading

File: _docs_integrate/version-5.md (hunk 2 above)
Comment (Magnus-Kuhn) [reply _docs_integrate/version-5.md:28]: didn't know it wasn't necessary :)

Comment (jkoenig134) [reply _docs_integrate/version-5.md]: it is necessary if you want to rewrite the anchor from `#removalofrelationshipchanges` to `#relationshipchanges` which we don't want

File: _docs_integrate/version-5.md (hunk 1 above)
Comment (jkoenig134) [reply _docs_integrate/version-5.md:102]: So you want to talk about relationships in the title instead of the creationContent inside? I'm ok with that.

File: _docs_integrate/integration-example.md [hunk 3]
```diff
@@ -222,23 +222,22 @@ To do so, we [synchronize updates of the Backbone]({% link _docs_use-cases/use-c
 
 {% include rapidoc api_route_regex="^post /api/v2/Account/Sync$" %}
 
-In the Response we will receive the Relationship.
+We can now [get the Relationship]({% link _docs_use-cases/use-case-transport-query-relationships.md %}).
+
+{% include rapidoc api_route_regex="^get /api/v2/Relationships$" %}
```
Comment (jkoenig134) [inline _docs_integrate/integration-example.md:227]: maybe you should explain

1. how to filter for the new relationship for example by the template
2. also say that you should listen for the event RelationshipChanged

Comment (jkoenig134) [reply _docs_integrate/integration-example.md:227]: b/c we don't want the people to query here, instead to listen to events

File: _docs_integrate/version-5.md [hunk 4]
```diff
@@ -99,6 +99,69 @@ In addition, in contrast to the RequestItems, a RequestItemGroup did not have to
...
+for creating a Relationship via POST `Relationship`.
+
+Because previously a JSONWrapper was applied to `content: <arbitrary-content>` before actually sending it and the `content` property of the created Message resp. RelationshipTemplate then was
+
+```jsonc
+content: {
+  "@type": "JSONWrapper",
+  "value": <arbitrary-content>
+},
+```
```
Comment (jkoenig134) [inline _docs_integrate/version-5.md:157]: I think this is incorrect. JSONWrapper only exists internally and when making a JSONWrapper toJSON we only get the value back.

Comment (Magnus-Kuhn) [reply _docs_integrate/version-5.md:157]: good to know

File: _docs_integrate/integration-example.md (hunk 3 above)
Comment (Magnus-Kuhn) [reply _docs_integrate/integration-example.md:227]: We already have a disclaimer at the top to disable the Sync Module, I'll expand it. The Relationships and Messages tutorials will need some event adaptation, the requests tutorial only needs to be updated with the sse-Module.
I would like to do the eventing adaptions in a different PR though

**Review by britsta — COMMENTED** [review]
Please also mention the `ArbitraryRelationshipTemplateContent` in the [Establish Relationships](https://enmeshed.eu/integrate/establish-relationships) scenario documentation. For example, the header [RelationshipTemplate without RelationshipTemplateContent](https://enmeshed.eu/integrate/establish-relationships#relationshiptemplate-without-relationshiptemplatecontent) can be adapted there.

File: _docs_integrate/data-model-overview.md [hunk 5]
```diff
@@ -70,31 +70,31 @@ A RelationshipTemplate serves two purposes:
...
+| creationContent | [`RelationshipCreationContent`](#relationshipcreationcontent) \| [`ArbitraryRelationshipCreationContent`](#arbitraryrelationshipcreationcontent) | The content sent along when the Relationship is initiated. If the other party uses the enmeshed App, `RelationshipCreationContent` has to be used. The `ArbitraryRelationshipCreationContent` can be filled with anything and serves as a fallback if the `RelationshipCreationContent` is not sufficient.                                                                                                                                                     | will be encrypted before sent to the Backbone |
```
Comment (britsta) [inline _docs_integrate/data-model-overview.md:95]: As I understand it, the `creationContent` is automatically filled with a RelationshipCreationContent if the Relationship was initiated after answering a Request contained within a RelationshipTemplateContent. The wording here suggests as if you have to actively use the RelationshipCreationContent. 

Furthermore, in this case the initiator of the Relationship would be the App user and the other party would not necessarily be an App user, as formulated here.

File: _docs_use-cases/use-case-transport-create-relationship-with-relationshiptemplate.md [hunk 6]
```diff
@@ -42,12 +42,12 @@ api_route_regex: ^POST /api/v2/Relationships$
 
 {% include properties_list.html %}
 
-This use case intends to create a [Relationship]({% link _docs_integrate/data-model-overview.md %}#relationship) based on a received [RelationshipTemplate]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplate).
+This use case intends to create a [Relationship]({% link _docs_integrate/data-model-overview.md %}#relationship) based on a received [RelationshipTemplate]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplate). The Relationship will be with the Template's creator.
```
Comment (britsta) [inline _docs_use-cases/use-case-transport-create-relationship-with-relationshiptemplate.md:45]: Please write "RelationshipTemplate" instead of "Template" or use "Templator" to refer to the creator of the RelationshipTemplate as we do in other guides.

File: _docs_use-cases/use-case-transport-create-relationship-with-relationshiptemplate.md [hunk 7]
```diff
@@ -42,12 +42,12 @@ api_route_regex: ^POST /api/v2/Relationships$
...
 
 ## Parameters
 
 - `templateId` references the RelationshipTemplate that was received from a party.
-- `creationContent` can be used as a response with arbitrary data to the peer. This response is usually related to the data received by the RelationshipTemplate, e.g. the RelationshipTemplate contains a [Request]({% link _docs_integrate/data-model-overview.md %}#request) and the content here contains the respective [Response]({% link _docs_integrate/data-model-overview.md %}#response) to the Request.
+- `creationContent` is stored inside the Relationship. It's a [RelationshipCreationContent]({% link _docs_integrate/data-model-overview.md %}#relationshipcreationcontent) if the received Template contained a [RelationshipTemplateContent]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplatecontent), otherwise it is an [ArbitraryRelationshipCreationContent]({% link _docs_integrate/data-model-overview.md %}#arbitraryrelationshipcreationcontent). It can be used by the Template's creator to decide whether to accept the Relationship.
```
Comment (britsta) [inline _docs_use-cases/use-case-transport-create-relationship-with-relationshiptemplate.md:50]: As far as I know, this shouldn't be right. Only use this use case, if no RelationshipTemplateContent, but an ArbitraryRelationshipTemplateContent was used inside the RelationshipTemplate. When using a RelationshipTemplateContent, use the "Accept incoming Request" use case to initiate a Relationship or the "Reject incoming Request" use case to refuse the initiation of a Relationship. If an ArbitraryRelationshipTemplateContent is used, this use case can be called and an ArbitraryRelationshipCreationContent needs to be specified for the `creationContent` parameter here. 

This is not validated yet, but will be included in https://github.com/nmshd/runtime/pull/212 as this was an already known error-prone behavior.

File: _docs_use-cases/use-case-transport-synchronize-updates-of-backbone.md
Comment (britsta) [inline _docs_use-cases/use-case-transport-synchronize-updates-of-backbone.md:1]: A message further up on this page talks about "return values", which should no longer be up-to-date, as the use case no longer returns any content.

File: _docs_use-cases/use-case-transport-create-own-relationshiptemplate.md [hunk 8]
```diff
@@ -49,7 +49,7 @@ A [RelationshipTemplate]({% link _docs_integrate/data-model-overview.md %}#relat
 ## Parameters
 
 - `expiresAt` is the ISODateTime the RelationshipTemplate expires at. This should be set as small as possible and cannot be in the past.
-- `content` describes the structure of the RelationshipTemplate. Whereas the enmeshed App requires a [RelationshipTemplateContent]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplatecontent) as structure, the content of the RelationshipTemplate is arbitrary data.
+- `content` describes the structure of the RelationshipTemplate. The enmeshed App requires a [RelationshipTemplateContent]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplatecontent) as structure, otherwise an [ArbitraryRelationshipTemplateContent]({% link _docs_integrate/data-model-overview.md %}#arbitraryrelationshiptemplatecontent) containing any data can be used.
```
Comment (britsta) [inline _docs_use-cases/use-case-transport-create-own-relationshiptemplate.md:52]: Please be more precise about the fact that if the Identity **receiving** the RelationshipTemplate is an App user, the creator of the RelationshipTemplate must use a RelationshipTemplateContent.

File: _docs_integrate/establish-relationships.md [hunk 9]
```diff
@@ -139,18 +139,20 @@ Note that it is of course also possible to reject the incoming Request, if the R
 
 #### RelationshipTemplate without RelationshipTemplateContent
```
Comment (britsta) [inline _docs_integrate/establish-relationships.md]: Please write "RelationshipTemplate with ArbitraryRelationshipTemplateContent" here instead.

File: _docs_integrate/establish-relationships.md [hunk 10]
```diff
@@ -139,18 +139,20 @@ Note that it is of course also possible to reject the incoming Request, if the R
...
+We now consider the situation in which the [RelationshipTemplate]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplate) loaded onto the Requestor does not contain a RelationshipTemplateContent in its `content` property. In this case, the Requestor does not receive an incoming Request. Nevertheless, it can initiate a Relationship with the Templator by explicitly creating a data object of type [Relationship]({% link _docs_integrate/data-model-overview.md %}#relationship) with `"Pending"` as `status` based on the RelationshipTemplate. To do this, follow the instructions of the [Create Relationship with RelationshipTemplate]({% link _docs_use-cases/use-case-transport-create-relationship-with-relationshiptemplate.md %}) use case and provide
```
Comment (britsta) [inline _docs_integrate/establish-relationships.md:142]: Please extend the first sentence: "We now consider the situation in which the [RelationshipTemplate]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplate) loaded onto the Requestor does not contain a RelationshipTemplateContent in its `content` property, but an ArbitraryRelationshipTemplateContent." and link to the ArbitraryRelationshipTemplateContent.

File: _docs_use-cases/use-case-transport-synchronize-updates-of-backbone.md
Comment (Magnus-Kuhn) [reply _docs_use-cases/use-case-transport-synchronize-updates-of-backbone.md:1]: That PR isn't merged yet

File: _docs_use-cases/use-case-transport-create-relationship-with-relationshiptemplate.md (hunk 6 above)
Comment (Magnus-Kuhn) [reply _docs_use-cases/use-case-transport-create-relationship-with-relationshiptemplate.md:45]: Templator is not used in the use cases though

File: _docs_use-cases/use-case-transport-synchronize-updates-of-backbone.md
Comment (britsta) [reply _docs_use-cases/use-case-transport-synchronize-updates-of-backbone.md:1]: Which PR are you referring to? I am not sure if I understand you correctly, because in this file `use-case-transport-synchronize-updates-of-backbone.md` it is already written under section "On Success" that in case of success no content is returned.

Comment (Magnus-Kuhn) [reply _docs_use-cases/use-case-transport-synchronize-updates-of-backbone.md:1]: Sorry, mixed things up, you're totally right

File: _docs_integrate/version-5.md (hunk 1 above)
Comment (Magnus-Kuhn) [reply _docs_integrate/version-5.md:102]: Do you see anything left to do here?

File: _docs_integrate/read-attributes-from-peer.md
Comment (britsta) [inline _docs_integrate/read-attributes-from-peer.md:1]: I think in this file a merge conflict has led to mistakes and all changes made here can be discarded. 

(The deleted `{: .notice--info}` must actually remain in this file, and the added paragraphs can be removed as they are already included directly below.)

File: _docs_integrate/version-5.md [hunk 11]
```diff
@@ -99,6 +99,60 @@ In addition, in contrast to the RequestItems, a RequestItemGroup did not have to
...
+
+## Backbone-Sync returns 204 (No content)
+
+Previosly, the Backbone-Sync (via the route POST `Account/Sync`) returned a list of Relationships and Messages with the HTTP code 201. This has been changed to the sync returning nothing with the HTTP code 204. This removes the support for the workflow of regularly calling the sync and checking the response since unexpected response values happen if syncs are executed from different spots. We recommend working event-based with the [Message Broker Publisher-Module]({% link _docs_operate/modules.md %}#messagebrokerpublisher), which sends events to a message broker you have set up. We send a variety of events like `incomingRequestReceived`, `incomingRequestStatusChanged`, each of them containing the relevant changed object. See [Connector Events] for the full list. Slightly related, we also recommend using the recently published [server-sent-events Module]{% link _docs_operate/modules.md %}#sse, which listens to events emitted from the Backbone and syncs with the Backbone when an event is received.
```
Comment (britsta) [inline _docs_integrate/version-5.md:154]: "Previously" instead of "Previosly".

Comment (britsta) [inline _docs_integrate/version-5.md:154]: Fix the link "[server-sent-events Module]{% link _docs_operate/modules.md %}#sse" please, brackets are missing there.

File: _docs_integrate/version-5.md [hunk 12]
```diff
@@ -99,6 +99,60 @@ In addition, in contrast to the RequestItems, a RequestItemGroup did not have to
...
+and the parameter
+
+```jsonc
+creationContent: {
+  "@type": "ArbitraryRelationshipCreationContent",
+  "value": <arbitrary-content>
+}
+```
+
+for creating a Relationship via POST `Relationship`.
```
Comment (britsta) [inline _docs_integrate/version-5.md:148]: If we mention the routes instead of use cases, then please use the complete `POST /api/v2/Relationships`.

File: _docs_integrate/version-5.md [hunk 13]
```diff
@@ -99,6 +99,60 @@ In addition, in contrast to the RequestItems, a RequestItemGroup did not have to
...
+Thus, when sending a Message with non-standard content via POST `Messages`, instead of sending the parameter `content : <arbitrary-content>`, you send
+
+```jsonc
+content: {
+  "@type": "ArbitraryMessageContent",
+  "value": <arbitrary-content>
+}
+```
+
+Similarly, when creating a RelationshipTemplate with non-standard content via POST `RelationshipTemplates`, you send the parameter
```
Comment (britsta) [inline _docs_integrate/version-5.md:130]: If we mention the routes instead of use cases, then please use the complete `POST /api/v2/RelationshipTemplates/Own`.

File: _docs_integrate/version-5.md [hunk 14]
```diff
@@ -99,6 +99,60 @@ In addition, in contrast to the RequestItems, a RequestItemGroup did not have to
...
+Those new wrappers all have the type
+
+```jsonc
+{
+  "@type": "Arbitrary<...>Content",
+  "value": any,
+}
+```
+
+Thus, when sending a Message with non-standard content via POST `Messages`, instead of sending the parameter `content : <arbitrary-content>`, you send
```
Comment (britsta) [inline _docs_integrate/version-5.md:121]: If we mention the routes instead of use cases, then please use the complete `POST /api/v2/Messages`.

Comment (Magnus-Kuhn) [reply _docs_integrate/version-5.md:121]: Actually I'm not sure if the use cases or the routes are better, I guess it depends on how it's introduced

Comment (jkoenig134) [reply _docs_integrate/version-5.md]: routes make these kinds of guide easier to process. Please provide routes at minimum (in the format @britsta mentioned) and then if you want additionally the usecases

File: _docs_integrate/data-model-overview.md [hunk 15]
```diff
@@ -353,7 +353,6 @@ In the [Attribute introduction]({% link _docs_integrate/attribute-introduction.m
...
-| isDefault    | `true` \| `undefined`                                                                          | States whether the LocalAttribute is the default RepositoryAttribute for its Attribute value type. For every IdentityAttribute value type exactly one RepositoryAttribute has `isDefault` set, given that at least one RepositoryAttribute of that value type exists. This property is only used for the App, e.g. to mark the Attribute that should be used by default, if multiple Attributes of the same value type exist. Thus, for a Connector it will always be `undefined`.                                                                 |
```
Comment (britsta) [inline _docs_integrate/data-model-overview.md:356]: Why was this removed?

Comment (Magnus-Kuhn) [reply _docs_integrate/data-model-overview.md]: Idk, it's back

File: _docs_integrate/version-5.md (hunk 1 above)
Comment (britsta) [reply _docs_integrate/version-5.md]: FYI, the PR has now been taken over by me @jkoenig134 :D This comment can be resolved, right?

Comment (jkoenig134) [reply _docs_integrate/version-5.md]: yep

**Review by jkoenig134 — APPROVED** [review]
